include/yk/compare/comparator.hpp

```diff
@@ -20,18 +20,12 @@ namespace compare {
 namespace detail {
 
 // is there more better way to detect range adaptor closure? idk
-template <class T>
-inline constexpr bool is_range_adaptor_closure_v =
-#if defined(__GLIBCXX__)
-    std::views::__adaptor::__is_range_adaptor_closure<T>
-#elif defined(_LIBCPP_VERSION)
-    std::ranges::_RangeAdaptorClosure<T>
-#elif defined(_MSC_VER)
-    std::ranges::_Pipe::_Range_adaptor_closure_object<T>
-#else
-    false
-#endif
-    ;
+template <class Closure>
+concept RangeAdaptorClosure = requires(Closure closure) {
+  closure(std::views::empty<int>);
```

Comment:
Instantiating the specific type `std::views::empty<int>` for a more generic concept seems not elegant. I think we should define a minimal class that models a range, and use it solely as our internal type.

```cpp
namespace detail {
struct just_a_range : ... { ... }; // For example. You can make a decision on the class name; choose it wisely.
}
```